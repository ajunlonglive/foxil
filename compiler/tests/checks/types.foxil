@AliasToUnknown = type @Unknown
@AliasToSelf = type @AliasToSelf

@FastAlias = type { i32 %keep }

@Zero = type {}

func @main() void {
    %q = alloca { i32 %f, bool %f2 }, { i32 %f, bool %f2 } { i32 5, bool true, char 'A' }
    %q2 = alloca { bool %f2 }, { bool %f2 } { i32 5, bool true, char 'A' }
    
    %f = alloca @FastAlias, @FastAlias { i32 4 }
    %f2 = alloca @FastAlias, @FastAlias { }
    
    %z = alloca @Zero, @Zero {i32 0}
    ret void
}
