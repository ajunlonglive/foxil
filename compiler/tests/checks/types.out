compiler/tests/checks/types.foxil:1:23: error: type `Unknown` not found
     1 | @AliasToUnknown = type @Unknown
       |                        ^^^^^^^^
compiler/tests/checks/types.foxil:2:1: error: a type alias cannot refer to itself
     2 | @AliasToSelf = type @AliasToSelf
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
compiler/tests/checks/types.foxil:9:38: error: too few expressions to type ‘{ i32, bool }’ (2 expression(s) are expected, not 3)
     9 |     %q = alloca { i32 %f, bool %f2 }, { i32 %f, bool %f2 } { i32 5, bool true, char 'A' }
       |                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
compiler/tests/checks/types.foxil:10:31: error: too few expressions to type ‘{ bool }’ (1 expression(s) are expected, not 3)
    10 |     %q2 = alloca { bool %f2 }, { bool %f2 } { i32 5, bool true, char 'A' }
       |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
compiler/tests/checks/types.foxil:13:29: error: too many expressions to type ‘{ i32 }’ (1 expression(s) are expected, not 0)
    13 |     %f2 = alloca @FastAlias, @FastAlias { }
       |                              ^^^^^^^^^^^^^^
compiler/tests/checks/types.foxil:15:23: error: too few expressions to type ‘{  }’ (0 expression(s) are expected, not 1)
    15 |     %z = alloca @Zero, @Zero {i32 0}
       |                        ^^^^^^^^^^^^^
